google/cloud/logging_v2/logger.py

```diff
@@ -457,6 +461,36 @@ def commit(self, *, client=None):
             kwargs["labels"] = self.logger.labels
 
         entries = [entry.to_api_repr() for entry in self.entries]
-
-        client.logging_api.write_entries(entries, **kwargs)
+        try:
+            client.logging_api.write_entries(entries, partial_success=True, **kwargs)
+        except InvalidArgument as e:
+            # InvalidArgument is often sent when a log is too large
+            # attempt to attach extra contex on which log caused error
+            self._append_context_to_error(e)
+            raise e
         del self.entries[:]
+
+    def _append_context_to_error(self, err):
+        """
+        Attempts to Modify `write_entries` exception messages to contain
+        context on which log in the batch caused the error.
+
+        Best-effort basis. If another exception occurs while processing the
+        input exception, the input will be left unmodified
+
+        Args:
+            err (~google.api_core.exceptions.InvalidArgument):
+                The original exception object
+        """
+        try:
+            # find debug info proto if in details
+            debug_info = next(x for x in err.details if isinstance(x, DebugInfo))
```

Comment:
Is there any public reference on that debug info structure?

Reply:
The structure is all documented as part of the [InvalidArgument exception](https://googleapis.dev/python/google-api-core/latest/exceptions.html#google.api_core.exceptions.InvalidArgument) (which was listed in the Args section above)